.github/workflows/define-matrix.yml

```diff
@@ -38,5 +43,23 @@ jobs:
       - name: Generate matrix data
         id: get_matrix
         run: |
-          MATRIX=$(python3 scripts/define-matrix.py "${{ inputs.target_branch }}" "${{ inputs.use_full_releases }}")
+          # Build command arguments array
+          args=("python3" "scripts/define-matrix.py")
+
+          # Add target branch if specified
+          if [ -n "${{ inputs.target_branch }}" ]; then
+            args+=("${{ inputs.target_branch }}")
+          fi
+
+          # Add use_full_releases flag
+          if [ "${{ inputs.use_full_releases }}" = "false" ]; then
```

Comment:
I am still not sure what this is supposed to be doing - please add some explanation either here or where the argument is parsed, ideally with some examples. Thanks! :)